drivers/watchdog/wdt_iwdg_stm32.c

```diff
@@ -60,7 +69,7 @@ static void iwdg_stm32_convert_timeout(uint32_t timeout,
 	uint8_t shift = 0U;
 
 	/* Convert timeout to LSI clock ticks. */
-	uint32_t ticks = (uint64_t)timeout * LSI_VALUE / USEC_PER_SEC;
+	uint32_t ticks = (uint64_t)timeout * STM32_MAX_LSI_FREQ / USEC_PER_SEC;
```

Comment:
I'm still wondering a bit if this is what the user would expect. If a user reads the datasheet and already considers the tolerance in the timeout selection, he/she would get a significantly longer timeout than expected (on average).

Did you consider adjusting below Kconfig instead to make the task watchdog pass tests for the mentioned board?

https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/task_wdt/Kconfig#L47-L60

Reply:
absolutely, adjusting the  CONFIG_TASK_WDT_HW_FALLBACK_DELAY=100  for the boards can also fix that

Reply:
The TASK_WDT_MIN_TIMEOUT is 1..10000, with a LSI clock variation from 37000 (typical) to 56000 (max), the CONFIG_TASK_WDT_HW_FALLBACK_DELAY cab be given by :

-    _upper value of TASK_WDT_MIN_TIMEOUT * LSI_CLOCK typical freq / USEC_PER_SEC - upper value of TASK_WDT_MIN_TIMEOUT * LSI_CLOCK max freq / USEC_PER_SEC_

depending on the soc: 
stm32F0x : CONFIG_TASK_WDT_HW_FALLBACK_DELAY=100
stm32F4x : CONFIG_TASK_WDT_HW_FALLBACK_DELAY=200
stm32L1x : CONFIG_TASK_WDT_HW_FALLBACK_DELAY=200
stm32L0x : CONFIG_TASK_WDT_HW_FALLBACK_DELAY=200